src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -64,7 +66,11 @@ impl IndexScanExecutor {
             pk_col = Some(cols.pop().unwrap());
         }
         let col_ids = cols.iter().map(|c| c.get_column_id()).collect();
-
+        let counts = if collect {
+            Vec::with_capacity(key_ranges.len())
+        } else {
+            Vec::default()
+        };
```

Comment:
Maybe you can define the `counts` as an Option? It could bring the following benefits:
1. Make it easier to check whether we need to collect
2. in `collect_output_counts `, you can use `take` instead of `drain`

Reply:
I think that it is better to not do this because the `counts ` is initialized with capacity, after `take`, it still needs to `drain`.